lib/instrumentation/index.js

```diff
@@ -216,7 +226,8 @@ Instrumentation.prototype._startHook = function () {
     var enabled = self._agent._conf.instrument && !disabled.has(name)
     var pkg, version
 
-    if (basedir) {
+    const lambdaHandlerInfo = getLambdaHandler(process.env)
+    if (!lambdaHandlerInfo && basedir) {
```

Comment:
This logic feels wrong. Doesn't this change the version for, e.g., 'express'? I.e. for *all* non-core modules.

Reply:
Right you are -- looks like this inadvertently just checks for the existence of the handler instead of checking for both the existence of the handler AND whether the instrumentation currently being handled is the lambda instrumentation.  Test added to make sure the existing instrumentation are applied when lambda's detected and the correct (one hopes) logic's been added here: https://github.com/elastic/apm-agent-nodejs/pull/2577/commits/2e38f10fc9c4c4e6777af5fe6a74129cd411cd78#diff-413f1f1acaa8b1d97f8cae0884c443a777ef6eb822092e6483b4fe59a4936221R108